sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java

```diff
@@ -0,0 +1,681 @@
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read from the HBase instance
+         * indicated by the given configuration.
+         */
+        public Read withConfiguration(Configuration configuration) {
+            checkNotNull(configuration, "conf");
+            return new Read(new SerializableConfiguration(configuration),
+                    tableId, serializableScan);
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read from the specified table.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withTableId(String tableId) {
+            checkNotNull(tableId, "tableId");
+            return new Read(serializableConfiguration, tableId, serializableScan);
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will filter the rows read from HBase
+         * using the given scan.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withScan(Scan scan) {
+            checkNotNull(scan, "scan");
+            return new Read(serializableConfiguration, tableId, new SerializableScan(scan));
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will filter the rows read from HBase
+         * using the given row filter.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withFilter(Filter filter) {
+            checkNotNull(filter, "filter");
+            return withScan(serializableScan.getScan().setFilter(filter));
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read only rows in the specified range.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withKeyRange(ByteKeyRange keyRange) {
+            checkNotNull(keyRange, "keyRange");
+            byte[] startRow = keyRange.getStartKey().getBytes();
+            byte[] stopRow = keyRange.getEndKey().getBytes();
+            return withKeyRange(startRow, stopRow);
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read only rows in the specified range.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withKeyRange(byte[] startRow, byte[] stopRow) {
+            checkNotNull(startRow, "startRow");
+            checkNotNull(stopRow, "stopRow");
+            return withScan(serializableScan.getScan().setStartRow(startRow).setStopRow(stopRow));
+        }
+
+        private Read(SerializableConfiguration serializableConfiguration, String tableId,
+                     SerializableScan serializableScan) {
+            this.serializableConfiguration = serializableConfiguration;
+            this.tableId = tableId;
+            this.serializableScan = serializableScan;
+        }
+
+        @Override
+        public PCollection<Result> expand(PBegin input) {
+            HBaseSource source = new HBaseSource(this, null);
```

Comment:
ok, again idea copied from BigtableSource that does not do this. I will fix as you proposed.